ui/app/services/system.js

```diff
@@ -39,7 +39,12 @@ export default class SystemService extends Service {
         .authorizedRawRequest(`/${namespace}/agent/self`)
         .then(jsonWithDefault({}))
         .then(agent => {
-          agent.version = agent.member?.Tags?.build || 'Unknown';
+          if (agent.config.Version) {
+            const { Version, VersionPrerelease, VersionMetadata } = agent.config.Version;
+            agent.version = Version;
+            if (VersionPrerelease) agent.version += `-${VersionPrerelease}`;
+            if (VersionMetadata) agent.version += `+${VersionMetadata}`;
```

Comment:
Little quibble/nit here, maybe use 

```javascript
agent.version = `${agent.version}-${VersionPrerelease}`
```

etc if you are using template literals anyway?

To me it reads a little easier and personally I like using template strings instead of using `+`/`+=` for string concatenation. Complete nit though, so I'll leave it up to you.

Reply:
We're conditionally adding to `agent.version` because `Prerelease` and `Metadata` can be empty strings.

So would the better way to do this be:

```
agent.version = Version;
if (VersionPrerelease) {
  agent.version = `${agent.version}-${VersionPrerelease}`
}
if (VersionMetadata) {
  agent.version = `${agent.version}-${MetaData}`
}
```